sunpy/util/tests/test_config.py

```diff
@@ -80,3 +86,17 @@ def test_get_and_create_sample_dir():
     # Set the config back
     os.rmdir(new_path)
     config.set('downloads', 'sample_dir', os.path.join(USER, 'sunpy', 'data', 'sample_data'))
+
+
+def test_write_config():
+    config_filename = 'sunpyrc'
+    module_dir = Path(sunpy.__file__).parent
+    config_file = module_dir / 'data' / config_filename
+    user_config_dir = Path(_get_user_configdir())
+    user_config_file = user_config_dir / config_filename
+
+    write_config()
+    assert open(user_config_file, 'r').read() == open(config_file, 'r').read()
+
+    with pytest.warns(SunpyUserWarning):
+        write_config(overwrite=False)
```

Comment:
Does this test use a real user's directory?

Reply:
Yes. This is a test to check if the default sunpy config file is copied to the user's config directory.

Reply:
We can't have a test that actually interferes with a running user's system. 

Can you check this does not affect an existing config file.

Reply:
It would. That was the whole point of the write function in the original PR. To copy the default config to the user config file. Create one if there isn't any. Overwrite it if the user wishes so.

Reply:
Yes but we can't have a test actually overwrite a real users config.

Reply:
Should I make a temporary folder and use it to test everything?

Reply:
Yes, override the config location, create a fake file and see if it overwrites and all that jazz. There are other config tests around that have done the first bit before.

Reply:
There's this problem that the `write_config` function by default writes in the user's config location.
I can override it by passing a parameter to the function to allow user-specified locations. But would such an option to write config files wherever the user wants to, make sense?

Reply:
It would be better to change the location inside the test only.